play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/implicits/RichDateInputSupport.scala

```diff
@@ -65,35 +145,72 @@ trait RichDateInputSupport {
+        deprecatedInputItem(field, dateInputItems(1), "month", className = "govuk-input--width-2"),
+        deprecatedInputItem(field, dateInputItems(2), "year", className = "govuk-input--width-4")
+      )
+
+      dateInput.copy(items = items)
+    }
+
+    private def deprecatedInputItem(field: Field, inputItem: InputItem, key: String, className: String): InputItem = {
       def errorClass(itemField: Field) =
         if (field.errors.nonEmpty || itemField.errors.nonEmpty) "govuk-input--error" else ""
 
-      def inputItem(inputItem: InputItem, key: String, className: String): InputItem = {
-        val defaultInputItem = InputItem.defaultObject
-        val classes          = if (inputItem.classes == defaultInputItem.classes) className else inputItem.classes
+      val defaultInputItem = InputItem.defaultObject
+      val classes          = if (inputItem.classes == defaultInputItem.classes) className else inputItem.classes
 
-        inputItem.copy(
-          id = if (inputItem.id == defaultInputItem.id) s"${field.name}.$key" else inputItem.id,
-          name = if (inputItem.name == defaultInputItem.name) s"${field.name}.$key" else inputItem.name,
-          value = if (inputItem.value == defaultInputItem.value) field(key).value else inputItem.value,
-          label =
-            if (inputItem.label == defaultInputItem.label) Some(messages(s"date.input.$key")) else inputItem.label,
-          classes = s"$classes ${errorClass(field(key))}".trim
-        )
-      }
+      inputItem.copy(
+        id = if (inputItem.id == defaultInputItem.id) s"${field.name}.$key" else inputItem.id,
+        name = if (inputItem.name == defaultInputItem.name) s"${field.name}.$key" else inputItem.name,
+        value = if (inputItem.value == defaultInputItem.value) field(key).value else inputItem.value,
+        label = if (inputItem.label == defaultInputItem.label) Some(messages(s"date.input.$key")) else inputItem.label,
+        classes = s"$classes ${errorClass(field(key))}".trim
+      )
+    }
 
-      val dateInputItems = if (dateInput.items.size == 3) dateInput.items else Seq.fill(3)(InputItem.defaultObject)
+    private def defaultInputItem(field: Field, key: String, className: String): InputItem = {
+      def errorClass(itemField: Field) =
+        if (field.errors.nonEmpty || itemField.errors.nonEmpty) "govuk-input--error" else ""
 
-      val items = Seq(
-        inputItem(dateInputItems(0), "day", className = "govuk-input--width-2"),
-        inputItem(dateInputItems(1), "month", className = "govuk-input--width-2"),
-        inputItem(dateInputItems(2), "year", className = "govuk-input--width-4")
+      InputItem(
+        id = s"${field.name}.$key",
+        name = s"${field.name}.$key",
+        value = field(key).value,
+        label = Some(messages(s"date.input.$key")),
+        classes = s"$className ${errorClass(field(key))}".trim
       )
```

Comment:
we've lost the ability here to pass existing values for InputItems, and I can't remember if this was a deliberate choice - this might impact users if they need to set any custom attributes, they might not be able to just swap from `withFormField` to `withDayMonthYearInputItems`

I wonder if this is a good chance to add in some `require()` guardrails around stuff since it's new api - so for example, if we stick with throwing away anything in existing items, then we could require items be empty, if we keep the existing items then we could do some checks like - with the current one we use the inputItem.name if it exists, except we don't check if `InputItem.name == "${field.name}.$key"` so theoretically someone might not realise that errors weren't being wired up as they expected if their input items weren't in the "${field.name}.(day|month|year)" format

not 100% on it - I might be sending us on a side quest

Reply:
My recall of this was that we're deliberately simplifying things so that teams can no longer pass through custom values to the `with...(field: Field)` helpers, as the idea was to make the defaults much simpler to use, then if people want to customise they can use their own. I do agree though that it is a change, and the idea of using a `@require` is a good one to make this super visible.

Reply:
@oscarduignan Added `require` to the helper methods.

Reply:
🙌

Reply:
looks good to me, that means that people won't get caught out when they move over